handbook/openid-connect/openid-connect.md

```diff
@@ -0,0 +1,133 @@
+ "ni-attributes": {
+   "displayName": "Log in with PingFederate",
+   "iconUri": "/login/assets/pf.png"
+ }
+}
+```
+In this example the `openid`, `email`, and `profile` scopes are requested. Additional scopes may be requested. Consult your provider's documentation regarding exposing scopes to clients. Each scope will contain claims that can be used to map to roles within workspaces in SystemLink. See [**Mapping OpenID Connect Claims to SystemLink Workspaces and Roles**](#mapping-openid-connect-claims-to-systemlink-workspaces-and-roles) for details. 
+
+The `ni-attributes` section determines the text and (optionally) an icon to be shown in the SystemLink login page. The `iconUri` is relative to `htdocs` directory (`C:\Program Files\National Instruments\Shared\Web Server\htdocs`)
+
+![SystemLink login windows with SSO login option](login-window.png)
+*SystemLink login windows with SSO login option. An icon has not been set in this example.*
+
+### ClientID and Secret 
+The `[provider-dns].client` file is used by the NI Web Server to authenticate with the provider. 
+
+```json
+{
+ "client_id": "slserver",
+ "client_secret": "4vFm89u07xaredactedredactedredactede2tjtsEGQhlLreLVjcyLA0"
+}
+
+```
+The `client_id` and `client_secret` can be obtained directly from the provider. Depending on the provider the `client_id` may be user defined. After setting these properties the NI Web Server must be restarted for changes to take effect. 
+
+#### ClientID and Secret Provider Documentation
+
+[PingFederate - Configuring an OAuth Client](https://docs.pingidentity.com/bundle/pingfederate-93/page/roj1564002966901.html)
+
+[Using OAuth 2.0 to Access Google APIs](https://developers.google.com/identity/protocols/oauth2)
+
+[Okta - Find your Application's credentials](https://developer.okta.com/docs/guides/find-your-app-credentials/findcreds/)
+
+[ADFS - Build a server side application using OAuth confidential clients with AD FS 2016 or later](https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/development/enabling-oauth-confidential-clients-with-ad-fs)
+
+### OpenID Configuration and Discovery
+The `[provider-dns].provider` file includes the contents of the provider OpenID Connect configuration. This file informs SystemLink of of the various endpoints exposed by the provider that are used during the login process. 
+
+You may use curl to create this file. Replace `[provider-dns]` with the DNS of your OpenID Connect Provider. 
+```bash
+curl https://[provider-dns]/.well-known/openid-configuration -o [provider-dns].provider
+```
+
+### Setting Login Redirect URI
+The client configuration for your provider requires a redirect URL that is during the login flow. This must include the protocol (`https://` or `http:// `) and the DNS name of the SystemLink server. If the SystemLink server's DNS changes, this setting must be updated with the provider. 
+
+#### The SystemLink login redirect URL
+```
+[protocol]://[systemlink-dns]/login/openidc-redirect
+
+```
+
```

Comment:
Were you going to add something about the id_token encryption/signing settings that we've tested?

Reply:
Yes, this was added since you provided this comment.